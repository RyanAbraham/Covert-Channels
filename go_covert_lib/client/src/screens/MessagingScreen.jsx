import React from 'react';
import PropTypes from 'prop-types';
import FormControl from 'react-bootstrap/FormControl';
import Button from 'react-bootstrap/Button';

const MessagingScreen = (props) => {
  const {
    textToSend,
    setTextToSend,
    covertMessages,
    sendMessage,
  } = props;
  return (
    <div className="cc-messaging m-2">
      <h2 className="m-1">Messaging</h2>
      <FormControl
        as="textarea"
        className="cc-messaging__input w-75 m-1"
        value={textToSend}
        onChange={e => setTextToSend(e.target.value)}
      />
      <Button
        variant="primary"
        onClick={sendMessage}
        className="cc-messaging__send m-1"
        disabled={!textToSend}
      >
        Send Message
      </Button>
      <br />
      <div className="m-1">Incoming Messages</div>
      <FormControl
        as="textarea"
        className="cc-messaging__output w-75 m-1"
        value={covertMessages.join('\n')}
        readOnly
      />
    </div>
  );
};

MessagingScreen.propTypes = {
  textToSend: PropTypes.string.isRequired,
  setTextToSend: PropTypes.func.isRequired,
  covertMessages: PropTypes.array.isRequired,
  sendMessage: PropTypes.func.isRequired,
};

export default MessagingScreen;
