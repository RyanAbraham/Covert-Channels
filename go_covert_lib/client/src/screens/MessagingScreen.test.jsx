import React from 'react';
import { mount } from 'enzyme';
import MessagingScreen from './MessagingScreen';

describe('Messaging screen', () => {
  const covertMessages = [];
  const sendMessage = jest.fn();
  const textToSend = '';
  let msgScreen;
  const setTextToSend = jest.fn(val => msgScreen.setProps({ textToSend: val }));

  beforeEach(() => {
    msgScreen = mount(<MessagingScreen
      textToSend={textToSend}
      setTextToSend={setTextToSend}
      covertMessages={covertMessages}
      sendMessage={sendMessage}
    />);
  });

  afterEach(() => {
    msgScreen.unmount();
  });

  it('handles input in the input field', () => {
    let sendButtonIsDisabled = msgScreen.find('.cc-messaging__send').at(0).props().disabled;
    expect(sendButtonIsDisabled).toBe(true);
    // Simulate typing a character
    msgScreen.find('.cc-messaging__input').at(0).simulate('change', { target: { value: 'a' } });
    sendButtonIsDisabled = msgScreen.find('.cc-messaging__send').at(0).props().disabled;
    expect(setTextToSend).toHaveBeenCalled();
    expect(sendButtonIsDisabled).toBe(false);
    expect(msgScreen).toMatchSnapshot();
  });

  it('handles input submission', () => {
    msgScreen.find('.cc-messaging__input').at(0).simulate('change', { target: { value: 'a' } });
    const sendButton = msgScreen.find('.cc-messaging__send').at(0);
    sendButton.simulate('click');
    expect(sendMessage).toHaveBeenCalled();
  });

  it('displays incoming messages', () => {
    const msg1 = 'Coca-Cola';
    const msg2 = 'Never forget';
    msgScreen.setProps({ covertMessages: [msg1, msg2] });
    const outputField = msgScreen.find('.cc-messaging__output').at(0);
    expect(outputField.props().value).toEqual(`${msg1}\n${msg2}`);
    expect(msgScreen).toMatchSnapshot();
  });
});
