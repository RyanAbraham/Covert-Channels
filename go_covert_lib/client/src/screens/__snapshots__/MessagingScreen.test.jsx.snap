// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messaging Screen displays incoming messages 1`] = `
"<MessagingScreen textToSend=\\"\\" setTextToSend={[Function: mockConstructor]} covertMessages={{...}} sendMessage={[Function: mockConstructor]}>
  <div className=\\"cc-messaging m-2\\">
    <h2 className=\\"m-1\\">
      Messaging
    </h2>
    <FormControl as=\\"textarea\\" className=\\"cc-messaging__input w-75 m-1\\" value=\\"\\" onChange={[Function: onChange]}>
      <textarea value=\\"\\" onChange={[Function: onChange]} type={[undefined]} readOnly={[undefined]} id={[undefined]} className=\\"cc-messaging__input w-75 m-1 form-control\\" />
    </FormControl>
    <Button variant=\\"primary\\" onClick={[Function: mockConstructor]} className=\\"cc-messaging__send m-1\\" disabled={true} active={false} type=\\"button\\">
      <button onClick={[Function: mockConstructor]} disabled={true} type=\\"button\\" className=\\"cc-messaging__send m-1 btn btn-primary\\">
        Send Message
      </button>
    </Button>
    <br />
    <div className=\\"m-1\\">
      Incoming Messages
    </div>
    <FormControl as=\\"textarea\\" className=\\"cc-messaging__output w-75 m-1\\" value=\\"Coca-Cola\\\\nNever forget\\" readOnly={true}>
      <textarea value=\\"Coca-Cola\\\\nNever forget\\" type={[undefined]} readOnly={true} id={[undefined]} className=\\"cc-messaging__output w-75 m-1 form-control\\" />
    </FormControl>
  </div>
</MessagingScreen>"
`;

exports[`Messaging Screen handles input in the input field 1`] = `
"<MessagingScreen textToSend=\\"a\\" setTextToSend={[Function: mockConstructor]} covertMessages={{...}} sendMessage={[Function: mockConstructor]}>
  <div className=\\"cc-messaging m-2\\">
    <h2 className=\\"m-1\\">
      Messaging
    </h2>
    <FormControl as=\\"textarea\\" className=\\"cc-messaging__input w-75 m-1\\" value=\\"a\\" onChange={[Function: onChange]}>
      <textarea value=\\"a\\" onChange={[Function: onChange]} type={[undefined]} readOnly={[undefined]} id={[undefined]} className=\\"cc-messaging__input w-75 m-1 form-control\\" />
    </FormControl>
    <Button variant=\\"primary\\" onClick={[Function: mockConstructor]} className=\\"cc-messaging__send m-1\\" disabled={false} active={false} type=\\"button\\">
      <button onClick={[Function: mockConstructor]} disabled={false} type=\\"button\\" className=\\"cc-messaging__send m-1 btn btn-primary\\">
        Send Message
      </button>
    </Button>
    <br />
    <div className=\\"m-1\\">
      Incoming Messages
    </div>
    <FormControl as=\\"textarea\\" className=\\"cc-messaging__output w-75 m-1\\" value=\\"\\" readOnly={true}>
      <textarea value=\\"\\" type={[undefined]} readOnly={true} id={[undefined]} className=\\"cc-messaging__output w-75 m-1 form-control\\" />
    </FormControl>
  </div>
</MessagingScreen>"
`;
